test/e2e/branding_test.go

```diff
@@ -0,0 +1,198 @@
+	}
+	originalConfigBrand := originalConfig.Spec.Customization.Brand
+	// Get the original Managed Config Map
+	originalManagedConfigMap, err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Get(consoleapi.OpenShiftConsoleConfigMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		t.Fatalf("could not get console-config configmap, %v", err)
+	}
+
+	return client, originalConfigBrand, originalManagedConfigMap
+}
+
+func cleanupBrandingTestCase(t *testing.T, client *testframework.Clientset, originalConfigBrand operatorsv1.Brand, originalManagedConfigMap *v1.ConfigMap) {
+
+	time.Sleep(1)
+
+	setOperatorConfigBrand(t, client, originalConfigBrand)
+
+	if originalManagedConfigMap.Name != "" {
+		originalManagedConfigMap.SetResourceVersion("0")
+	} else {
+		originalManagedConfigMap = createConsoleConfigMap(operatorsv1.BrandOKD)
+	}
+	err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Delete(consoleapi.OpenShiftConsoleConfigMapName, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("could not delete managed config map %s, %v", originalManagedConfigMap.GetName(), err)
+	}
+	_, err = client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Create(originalManagedConfigMap)
+	if err != nil {
+		t.Fatalf("could not reset managed config map %s, %v", originalManagedConfigMap.GetName(), err)
+	}
+
+}
+
+// Test setting brand using the operator config which overrides the default and managed configmap
+func TestOperatorConfigBranding(t *testing.T) {
+	client, originalConfigBrand, originalManagedConfigMap := setupBrandingTestCase(t)
+	defer cleanupBrandingTestCase(t, client, originalConfigBrand, originalManagedConfigMap)
+	// Set a temporary managed config to test it does not override the operator config values
+	_, err := updateOrCreateConsoleConfigMap(client, consoleapi.OpenShiftConfigManagedNamespace, createConsoleConfigMap(operatorsv1.BrandOKD))
+	if err != nil {
+		t.Fatalf("error: could not apply managed config map %v", err)
+	}
+
+	for _, expectedBrand := range validBrands {
+		t.Logf("update operator with %v", expectedBrand)
+		// helper to update the operator config
+		setOperatorConfigBrand(t, client, expectedBrand)
+		// now check if it has set the brand
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getConsoleBrand(t, client)
+			//gotBrand := getOperaterConfigBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+
+}
+
+// Test setting brand via the config map in the openshift-config-managed namespace, this requires the operator config not be set
+func TestBrandingFromManagedConfigMap(t *testing.T) {
+	client, originalConfigBrand, originalManagedConfigMap := setupBrandingTestCase(t)
+	defer cleanupBrandingTestCase(t, client, originalConfigBrand, originalManagedConfigMap)
+	// Set operator config to empty so it does not override the managed config map values
+	setOperatorConfigBrand(t, client, "")
```

Comment:
I think this test is to check:
- if `console-config` in `openshift-config-managed` does not exist, branding is `ocp`.
- if `console-config` in `openshift-config-managed` does exist,  branding is `okd`.

We don't need the `operator config` correct?  
the `console-config` in `openshift-config-managed` isn't a public API, it should never exist with a different value than `ocp`. We don't need to loop all the brands and change it many times.

Reply:
We just need to make sure the `operator-config` is not set to any value or that will override the managed config.